lambda/translcore.ml

```diff
@@ -64,6 +64,29 @@ let check_record_field_sort loc : Jkind.Sort.Const.t -> _ = function
   | Product _ -> ()
   | Base Void -> raise (Error (loc, Illegal_void_record_field))
 
+let field_offset_for_label lbl =
+  match lbl.lbl_repres with
+  | Record_boxed _
+  | Record_inlined (_, Constructor_uniform_value, Variant_boxed _)
+  | Record_inlined (_, Constructor_uniform_value, Variant_with_null) ->
+      lbl.lbl_pos
+  | Record_inlined (_, Constructor_uniform_value, Variant_extensible) ->
+      lbl.lbl_pos + 1
+  | Record_unboxed | Record_inlined (_, _, Variant_unboxed) ->
+    (* For unboxed records, no offset calculation needed in regular field
```

Comment:
CR layouts 5.1

Reply:
I think I did what you wanted, but please double check?